apps/src/code-studio/components/progress/UnitCalendar.jsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import i18n from '@cdo/locale';
+import UnitCalendarLessonChunk from './UnitCalendarLessonChunk';
+import {unitCalendarLessonChunk} from '@cdo/apps/templates/progress/unitCalendarLessonShapes';
+
+const weekWidth = 592;
+const styles = {
+  weekColumn: {
+    minWidth: 100,
+    backgroundColor: 'rgb(118, 101, 160)',
+    color: 'white',
+    textAlign: 'center',
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    fontWeight: 'bold',
+    minHeight: 50
+  },
+  scheduleColumn: {
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    width: weekWidth,
+    display: 'flex',
+    minHeight: 50
+  },
+  table: {
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    width: '100%'
+  },
+  key: {
+    border: '1px solid rgb(118, 101, 160)',
+    borderCollapse: 'collapse',
+    width: '100%',
+    marginTop: 20
+  },
+  keyIcon: {
+    marginRight: 5
+  },
+  keySection: {
+    display: 'flex',
+    justifyContent: 'space-around',
+    width: '90%',
+    alignItems: 'center',
+    fontSize: 15
+  }
+};
+
+export default class UnitCalendar extends React.Component {
+  static propTypes = {
+    schedule: PropTypes.arrayOf(PropTypes.arrayOf(unitCalendarLessonChunk))
+      .isRequired,
+    weeklyInstructionalMinutes: PropTypes.number
+  };
+  constructor(props) {
+    super(props);
+    this.state = {
+      hovering: ''
+    };
+  }
+  handleHover = id => {
+    this.setState({hovering: id});
+  };
+  renderWeeks = () => {
+    const minuteWidth = weekWidth / this.props.weeklyInstructionalMinutes;
+    return this.props.schedule.map(week => {
+      return week.map((lesson, index) => {
+        return (
+          <UnitCalendarLessonChunk
+            key={`lesson-${index}`}
+            minuteWidth={minuteWidth}
+            lesson={lesson}
+            isHover={lesson.id === this.state.hovering}
+            handleHover={this.handleHover}
+          />
+        );
+      });
+    });
+  };
+
+  render() {
+    const weekPlans = this.renderWeeks();
+    return (
+      <div>
+        <table style={styles.table}>
+          <tbody>
+            {weekPlans.map((week, index) => {
+              return (
+                <tr key={`week-${index}`}>
+                  <td style={styles.weekColumn}>
+                    {i18n.weekLabel({number: index + 1})}
+                  </td>
+                  <td style={styles.scheduleColumn}>{week}</td>
+                </tr>
+              );
+            })}
+          </tbody>
+        </table>
+        <table style={styles.key}>
```

Comment:
This is a personal preference thing but you could consider making a UnitCalendarKey component to hold this information. Its not necessary at all its really a personal preference thing.

Reply:
I'm reluctant to duplicate the weekColumn and scheduleColumn styles, do you think it's worth putting just the keySection in it's own component?

Reply:
Nah I think its fine to leave it here.